tutorials/core-concepts/pieces/pieces-example.md

```diff
@@ -0,0 +1,242 @@
+      name: 'foods',
+      label: 'Food',
+      pluralLabel: 'Food',
+    
+      addFields: [
+      {
+        name: 'title',
+        label: 'Name',
+        type: 'string',
+        required: true
+      },
+      {
+        name: 'price',
+        label: 'Price',
+        type: 'float',
+        required: true
+      },
+      {
+        name: 'image',
+        label: 'Image',
+        type: 'singleton',
+        widgetType: 'apostrophe-images',
+        options: {
+          limit: 1,
+          minSize: [ 200, 200 ],
+          aspectRatio: [ 1, 1 ]
+      }
+     }
+     ]
+    };
+  ```
+
+
+```
+Note: Every time you update your module configuration in app.js you will need to restart Apostrophe for the changes to take effect.
+```
+
+## Create the Piece Widget
+
+Next, define the template for the widget to display the new pieces.
+
+1. Open `lib/modules/menu-item-widgets/views/widget.html`
+2. Add the following code to define the display template for the piece:
+  ```markup
+    {% for piece in data.widget._pieces %}
+
+      <h4>{% if piece._url %}<a href="{{ piece._url }}">{% endif %}
+
+      {{ piece.title }}
+
+      {% if piece._url %}</a>{% endif %}</h4>
+
+      ${{ piece.price }}
+
+      <p>{{ piece.description }}</p>
+
+      {{ apos.singleton( piece, 'image', 'apostrophe-images',
+
+        { edit: false }
+
+      )}}
+
+    {% endfor %}
+  ```
+
+## Displaying Pieces with a Widget
+
+Next, create some pieces to display and add the new widget to a page.
+
+### Create Pieces
+
+1. Log in to your Apostrophe instance.
+
+2. Open the main menu and select **Food**
+
+3. Create some food to be displayed.
+
+![](/.gitbook/assets/pieces-manage-food.png) 
+
+### Add the Widget to a Page
+
+1. Open `lib/modules/apostrophe-pages/views/pages/home.html`
+
+2. Add the widget to the template with an area
+
+  ```markup
+    {{ apos.area(data.page, 'body', {
+      widgets: {
+        'foods': {}
+        }
+      }
+    )}}
+  ```
+
+3. Navigate back to the home page, and you can now add a Menu Item Widget to the page.
+
+4. Click *Add Food Widget* and you can define which pieces to display and how to display them in the widget.
+
+![](/.gitbook/assets/pieces-add-food-widget.png) 
+
+```
+Note: in the widget template, you defined a **for loop** which looks through the list of all pieces to display. When you display the pieces in the widget, the selection you make in the dialog box will define the complete list of pieces that the template sees.
+```
+
+![](/.gitbook/assets/pieces-widget-page.png) 
+
+Congratulations! You created a new content type and a widget to display it. However, there are some limitations to using a widget to display this sort of content. One limitation is that everything you do and configure is contained within the instance of the widget placed on a specific page. Another is that the pieces themselves don’t have a “home” that you can universally link to them from. Using pieces pages will solve this.
```

Comment:
This paragraph is very clear, maybe it can be a source to improve the earlier paragraph that was confusing on this point.

Reply:
Another important limitation to be mentioned is that a widget is not a practical way to let users browse a thousand foods. Pieces pages are designed to support that use case through pagination and filtering.